.github/workflows/basic-tests-uv.yml

```
@@ -0,0 +1,66 @@
+name: Code tests (uv)
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - '**/*.py'
+      - '**/*.ipynb'
+      - '**/*.yaml'
+      - '**/*.yml'
+      - '**/*.sh'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - '**/*.py'
+      - '**/*.ipynb'
+      - '**/*.yaml'
+      - '**/*.yml'
+      - '**/*.sh'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  uv-tests:
+    name: UV Tests (Matrix OS)
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python (uv)
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install uv and dependencies
+        shell: bash
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
```

```suggestion
      - name: Install the latest version of uv
        uses: astral-sh/setup-uv@v5

      - name: Install dependencies
        shell: bash
        run: |
```

Unless there's a particular reason to manually install, using the Astral GitHub Action is faster.

---

Thanks! It's a bit of a weird case in this repo, but the reason why I am doing certain things manually is to test that this method / URL actually works. Like how a user would do it when reading the setup guides. Thanks for suggesting though!

---

Gotcha. Makes sense from a pedagogical perspective. That also explains the manual virtual env activation. :+1: